internal/controller/operator/factory/finalize/common.go

```diff
@@ -213,5 +214,28 @@ func FreeIfNeeded(ctx context.Context, rclient client.Client, object client.Obje
 	if err := RemoveFinalizer(ctx, rclient, object); err != nil {
 		return fmt.Errorf("cannot remove finalizer from object=%s/%s, kind=%q: %w", object.GetNamespace(), object.GetName(), object.GetObjectKind().GroupVersionKind(), err)
 	}
-	return fmt.Errorf("deletionTimestamp is not zero=%q for object=%s/%s kind=%s, recreating it at next reconcile loop. Warning never delete object manually", object.GetDeletionTimestamp(), object.GetNamespace(), object.GetName(), object.GetObjectKind().GroupVersionKind())
+	return &ErrWaitReady{
```

Comment:
This error must be returned to the user and it cannot be skipped. It indicates, that something is wrong and reconcile loop must be started again. 

 I think, instead of returning some wrapped error, it's better to check for exist finalizers and poll for object deletion:

```
	if object.GetDeletionTimestamp().IsZero() {
		// fast path
		return nil
	}
    if len(object.GetFinalizers) == 0 {
      wait.PollUntil(func{
           GetObject() == err.NotFound
     })
       return nil
   }

```c